mediapipe/tasks/ios/vision/core/sources/MPPTaskImage.h

```diff
@@ -0,0 +1,178 @@
+ * Hence, if this image is used as input for any MediaPipe vision tasks, inference will be performed
+ * on the it without any rotation. To create an `MPPTaskImage` with a different orientation, please
+ * use `[MPPTaskImage initWithImage:orientation:error:]`.
+ *
+ * @param image The image to use as the source. Its `CGImage` property must not be `NULL`.
+ * @param error An optional error parameter populated when there is an error in initializing the
+ * `MPPTaskImage`.
+ *
+ * @return A new `MPPTaskImage` instance with the given image as the source. `nil` if the given
+ * `image` is `nil` or invalid.
+ */
+- (nullable instancetype)initWithUIImage:(UIImage *)image
+                                   error:(NSError **)error NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Initializes an `MPPTaskImage` object with the given `UIImabe` and orientation.
+ *
+ * If the newly created `MPPTaskImage` is used as input for any MediaPipe vision tasks, inference
+ * will be performed on a copy of the image rotated according to the orientation.
+ *
+ * @param image The image to use as the source. Its `CGImage` property must not be `NULL`.
+ * @param orientation The display orientation of the image. This will be stored in the property
+ * `orientation`. `MPPTaskImage`.
+ * @param error An optional error parameter populated when there is an error in initializing the
+ * `MPPTaskImage`.
+ *
+ * @return A new `MPPTaskImage` instance with the given image as the source. `nil` if the given
+ * `image` is `nil` or invalid.
+ */
+- (nullable instancetype)initWithUIImage:(UIImage *)image
+                             orientation:(UIImageOrientation)orientation
+                                   error:(NSError **)error NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Initializes an `MPPTaskImage` object with the given pixel buffer.
+ *
+ * The orientation of the newly created `MPPTaskImage` will be `UIImageOrientationUp`.
+ * Hence, if this image is used as input for any MediaPipe vision tasks, inference will be performed
+ * on the it without any rotation. To create an `MPPTaskImage` with a different orientation, please
+ * use `[MPPTaskImage initWithPixelBuffer:orientation:error:]`.
+ *
+ * @param pixelBuffer The pixel buffer to use as the source. It will be retained by the new
+ *     `MPPTaskImage` instance for the duration of its lifecycle.
+ * @param error An optional error parameter populated when there is an error in initializing the
+ * `MPPTaskImage`.
+ *
+ * @return A new `MPPTaskImage` instance with the given pixel buffer as the source. `nil` if the
+ * given pixel buffer is `nil` or invalid.
+ */
+- (nullable instancetype)initWithPixelBuffer:(CVPixelBufferRef)pixelBuffer
+                                       error:(NSError **)error NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Initializes an `MPPTaskImage` object with the given pixel buffer and orientation.
+ *
+ * If the newly created `MPPTaskImage` is used as input for any MediaPipe vision tasks, inference
```

Comment:
@schmidt-sebastian @khanhlvg  I am hoping my understanding of the necessity of this property is correct. Please have a look.